src/libs/ReportUtils.js

```diff
@@ -259,6 +259,10 @@ function isCurrentUserSubmitter(reportID) {
  * @returns {Boolean}
  */
 function canDeleteReportAction(reportAction, reportID) {
+    const isActionOwner = reportAction.actorAccountID === currentUserAccountID;
+    if (isActionOwner && ReportActionsUtils.isMoneyRequestAction(reportAction) && !isSettled(reportAction.originalMessage.IOUReportID)) {
```

Comment:
In addition to checking if the report is not settled we should have also checked if the report is submitted. In paid policies you can't delete a money request in a submitted report (unless you are the admin).